### DISCUSSION TITLE
Casting UFix to SFix

### DISCUSSION THREAD

**Discussion by hmohajeri** [SpinalHDL/SpinalHDL#1609] [Q&A]
The following code does not compile due to fact that a is unsigned number and y is a signed number. How do I cast a UFix number to SFix? Many thanks.
class QCasing extends Component {
  val io = new Bundle {
    val x = in SFix(0 exp, -5 exp)  
    val y = out SFix(0 exp, -5 exp) 
  }

  val a = UFix(0 exp, -5 exp)
  a := 0.25

  io.y := a
}

showRtl( new QCasting)

**Comment by mrberman87** [comment]
The documentation hints to this: https://spinalhdl.github.io/SpinalDoc-RTD/master/SpinalHDL/Data%20types/Fix.html#type-cast

.toUFix() and .toSFix()

**Reply by hmohajeri** [reply]
Yes, I saw that before and tried it. I am getting the following message. Seems the library is missing isnide the 
cmd2.sc:11: value toSFix is not a member of spinal.core.UFix
  io.y := a.toSFix
            ^
Compilation Failed

and this is the simple code generating the error

class QCasting extends Component {
  val io = new Bundle {
    val x = in SFix(0 exp, -5 exp)  
    val y = out SFix(0 exp, -5 exp) 
    
  }

  val a = UFix(0 exp, -5 exp)
  a := 0.25

  io.y := a.toSFix
}


showRtl( new QCasting)
